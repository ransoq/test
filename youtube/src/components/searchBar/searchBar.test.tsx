import React from "react";
import SearchBar from "./searchBar";
import {shallow} from "enzyme";

describe("Testing <SearchBar />", () => {
    it("should save cookie after submit", () => {
        const value = "UCE9ODjNIkOHrnSdkYWLfYhg"
        const onSubmit = jest.fn();
        const wrapper = shallow(<SearchBar handleFormSubmit={onSubmit} />)

        expect(wrapper).toMatchSnapshot();

        wrapper.find('button').simulate('submit', {target: {value}});
        expect(onSubmit).toBe(value);
        Object.defineProperty(window.document, 'cookie', {
            writable: true,
            value: `lastFoundedChannel=UCE9ODjNIkOHrnSdkYWLfYhg`,
        });
    });
});