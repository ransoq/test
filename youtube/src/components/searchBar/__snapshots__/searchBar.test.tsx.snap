// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <SearchBar /> SearchBar have rendered correctly 1`] = `ShallowWrapper {}`;
